Robot2019/src/main/java/frc/robot/lib/logging/GlobalLogInfo.java

```diff
@@ -0,0 +1,92 @@
+package frc.robot.lib.logging;
+
+import java.io.File;
+
+import org.apache.commons.csv.CSVPrinter;
+
+/**
+ * Stores global information about the state of the logging code
+ */
+public final class GlobalLogInfo {
+
+    private static boolean isInit, areEventsDisabled, isDataDisabled;
+    private static File eventFile, dataFile;
+    private static CSVPrinter dataPrinter;
+
+    /**
+     * @return Whether the logging code has been initialized
+     */
+    public static boolean isInit() {
+        return isInit;
+    }
+
+    /**
+     * @return Whether event logging code has been disabled
+     */
+    public static boolean areEventsDisabled() {
+        return areEventsDisabled;
+    }
+
+    /**
+     * @return Whether data logging code has been disabled
+     */
+    public static boolean isDataDisabled() {
+        return isDataDisabled;
+    }
+
+    /**
+     * @return The {@link File} to which events are being logged
+     */
+    public static File getEventFile() {
+        return eventFile;
+    }
+
+    /**
+     * @return The {@link File} to which data is being logged
+     */
+    public static File getDataFile() {
+        return dataFile;
+    }
+
+    /**
+     * @return The {@link CSVPrinter} being used to log data
+     */
+    static CSVPrinter getDataPriter() {
+        return dataPrinter;
+    }
+
+    /**
+     * Disables event logging
+     */
+    static void disableEvents() {
+        areEventsDisabled = true;
+    }
+
+    /**
+     * Disables data logging
+     */
+    static void disableData() {
+        isDataDisabled = true;
+    }
+
+    /**
+     * Initializes the required global data as well as {@EventLog} and {@link DataLog} or returns if it has already been initialized
+     * @param eventFile The {@link File} to which events will be logged
+     * @param dataFile The {@link File} to which data will be logged
+     * @param dataPrinter The {@link CSVPrinter} to use to log data
+     */
+    static void init(File eventFile, File dataFile, CSVPrinter dataPrinter) {
+        if(isInit()) {
+            return;
+        }
+        GlobalLogInfo.eventFile = eventFile;
+        GlobalLogInfo.dataFile = dataFile;
+        GlobalLogInfo.dataPrinter = dataPrinter;
+        isInit = true;
+        EventLog.init();
+        DataLog.init();
```

Comment:
The EventLog and DataLog classes check the isInit flag to ensure that the required variables are initialized to complete initialization. The function will init them before returning, so unless we are multithreading, this will not cause any difference from outside of the logging code.

Reply:
I think I see your intention now. You're setting isInit = true to indicate that GlobalLogInfo's vars are inited. That's fine. However, EventLog.init() and DataLog.init() don't need to be called from GlobalLogInfo.init(). Those classes don't need to be inited for GlobalLogInfo to be inited. Also, it's not clear where GlobalLogInfo.init() is (or is supposed to be) called from. How about putting the following in Log.init():
GlobalLogInfo.init();
EventLog.init();
DataLog.init();

and removing the calls to EventLog.init() and DataLog.init() from this GlobalLogInfo.init().